internal/martian/proxy_handler.go

```diff
@@ -171,16 +171,13 @@ func (p proxyHandler) tunnel(name string, rw http.ResponseWriter, req *http.Requ
 		}
 		defer conn.Close()
 
-		if _, err := brw.WriteString(connectResponse); err != nil {
-			err := fmt.Errorf("got error while writing response back to client: %w", err)
-			p.traceWroteResponse(res, err)
-			return err
-		}
-		if err := brw.Flush(); err != nil {
-			err := fmt.Errorf("got error while flushing response back to client: %w", err)
-			p.traceWroteResponse(res, err)
-			return err
+		pc := proxyConn{
+			Proxy: p.Proxy,
+			brw:   brw,
+			conn:  conn,
 		}
+		pc.writeResponse(res)
```

Comment:
Fine, but it starts to look odd. Maybe there is some better way to share that code then 🤔.

Reply:
It looks ok to me tbh